### PR TITLE
Feat/postgre sql

### PR DESCRIPTION
**Pull request by hirichardhai**
working sub 7 minute data insert with PostgreSQL. please review my code. thanks!

### CODE REVIEW

**Comment by waynekwest88** [comment]
wow this is so impressive Richard!

File: db/SDC_postgreSQL/dataGenerator.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+const generateTenMilSimilarListings = async (collection, time) => {
+  for (let i = 0; i < 10000; i += 1) {
+    await batchInsertSimilarListings(i, collection);
+    console.log(`just finished inserting batch ${i}`);
+  }
+  const timeNow = new Date().getTime();
+  console.log(`${(timeNow - time) / 1000} seconds have passed`);
+};
+
```
Comment (kguinto) [inline db/SDC_postgreSQL/dataGenerator.js]: Good modularity in this file. 👍

File: db/SDC_postgreSQL/seed.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    data = [];
+    for (let i = 0; i < 10000; i += 1) {
+      const idx = (pageIndex * 10000) + i; // to insert unique product names
+      data.push(generateOneSimilarListing(idx));
+    }
+  }
+  return Promise.resolve(data);
+}
+
+console.log('starting to seed seed seedddddddd');
```
Comment (kguinto) [inline db/SDC_postgreSQL/seed.js]: lol

File: db/SDC_postgreSQL/dataGenerator.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  thumbnailimage: 'https://picsum.photos/' +
+                    `${settings.thumbnailImage.width}/` +
+                    `${settings.thumbnailImage.height}` +
+                    `?image=${id}`,
+});
+
+const insertIntoDatabase = (collection, array) => collection.insertMany(array);
+
+const batchInsertSimilarListings = (batchNumber, collection) => {
```
Comment (kguinto) [inline db/SDC_postgreSQL/dataGenerator.js]: Are you using these functions with postgres? These seem like they're left over from a Mongo implementation. If so, you might want to consider keeping different implementations on different files or different branches

File: db/SDC_postgreSQL/seed.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+const { generateOneSimilarListing } = require('./dataGenerator');
+
+const cn = {
+  host: 'localhost',
+  port: 5432,
+  database: 'seabnb',
+  user: 'student',
+  password: 'student',
+};
+
```
Comment (zinoki) [inline db/SDC_postgreSQL/seed.js]: might look cleaner if you eliminate spacing on variable declarations and keep them at the top

File: db/SDC_postgreSQL/seed.js (hunk 2 above)
Comment (zinoki) [inline db/SDC_postgreSQL/seed.js]: Do you need four console logs?

File: db/SDC_postgreSQL/dataGenerator.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  let similarListings = [];
+
```
Comment (zinoki) [inline db/SDC_postgreSQL/dataGenerator.js]: You could probably eliminate extra spaces

File: db/SDC_postgreSQL/dataGenerator.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+const pickRandomItem = array => array[getRandomNum(0, array.length)];
+
+const settings = {
+  similarListingsPerId: 10,
+
+  id: {
+    min: 0,
+    max: 200, // non-inclusive
+  },
+
```
Comment (zinoki) [inline db/SDC_postgreSQL/dataGenerator.js]: You have many spaces

**Comment by zinoki** [comment]
Looks good

File: db/SDC_postgreSQL/dataGenerator.js (hunk 3 above)
Comment (hirichardhai) [reply db/SDC_postgreSQL/dataGenerator.js]: good call, thanks!

File: db/SDC_postgreSQL/seed.js (hunk 2 above)
Comment (hirichardhai) [reply db/SDC_postgreSQL/seed.js]: yes, it looks cooler. 🥇

File: db/SDC_postgreSQL/dataGenerator.js (hunk 6 above)
Comment (hirichardhai) [reply db/SDC_postgreSQL/dataGenerator.js]: good eye

**Pull request merged**